astropy/units/quantity_helper.py

```diff
@@ -206,17 +206,20 @@ def helper_dimensionless_to_none(f, unit):
 
 
 def helper_power(f, unit1, unit2):
+    # TODO: find a better way to do this, currently need to signal that one
+    # still needs to raise power of unit1 in main code
     if unit2 is None:
-        return [None, None], _d(unit1)
+        return [None, None], False
 
-    # TODO: find a better way to do this, currently
-    # need to raise power of unit1 in main code
     try:
-        return [None, get_converter(unit2, dimensionless_unscaled)], _d(unit1)
+        return [None, get_converter(unit2, dimensionless_unscaled)], False
     except UnitsError:
         raise TypeError("Can only raise something to a dimensionless quantity")
 
 UFUNC_HELPERS[np.power] = helper_power
+# float_power was added in numpy 1.12
+if isinstance(getattr(np, 'float_power', None), np.ufunc):
```

Comment:
I'm not sure if it's really needed and works across numpy-versions but could you add a simple test which uses `float_power`?